client/src/partials/patient_invoice/registry/registry.js

```diff
@@ -0,0 +1,101 @@
+angular.module('bhima.controllers')
+.controller('InvoiceRegistryController', InvoiceRegistryController);
+
+InvoiceRegistryController.$inject = [
+  'PatientInvoiceService', '$uibModal', 'NotifyService', 'util', 'ReceiptModal'
+];
+
+/**
+ * Invoice Registry Controller
+ *
+ * This module is responsible for the management
+ * of Invoice Registry.
+ *
+ */
+function InvoiceRegistryController(Invoices, $uibModal, Notify, util, Receipt) {
+    var vm = this;
+
+    var invoiceActionsTemplate =
+        '<div style="padding : 5px"><a ng-click="grid.appScope.showBill(row.entity.uuid)">' +
+        '<span class="glyphicon glyphicon-list-alt"></span> {{ "TABLE.COLUMNS.BILL" | translate }}</a></div>';
+
+
+    vm.search = search;
+    vm.showBill = showBill;
+    vm.momentAge = util.getMomentAge;    
+
+    // track if module is making a HTTP request for invoices
+    vm.loading = false;
+
+    //setting columns names
+    vm.uiGridOptions = {
+        appScopeProvider : vm,
+        enableColumnMenus : false,
+        columnDefs : [
+            { field : 'reference', displayName : 'TABLE.COLUMNS.REFERENCE', headerCellFilter: 'translate' },
+            { field : 'date', cellFilter:'date', displayName : 'TABLE.COLUMNS.BILLING_DATE', headerCellFilter : 'translate' },
+            { field : 'patientNames', displayName : 'TABLE.COLUMNS.PATIENT', headerCellFilter : 'translate' },
+            { field : 'cost', cellFilter:'currency', displayName : 'TABLE.COLUMNS.COST', headerCellFilter : 'translate'  },
+            { field : 'serviceName', displayName : 'TABLE.COLUMNS.SERVICE', headerCellFilter : 'translate'  },
+            { field : 'createdBy', displayName : 'TABLE.COLUMNS.BY', headerCellFilter : 'translate' },
+            { name : 'Actions', displayName : '', cellTemplate : invoiceActionsTemplate }
+        ],
+        enableSorting : true
+    };
+
+    function handler(error) {
```

Comment:
If you click "Cancel" on the Search Modal, it will show an error occurred, when no error occurred.  See below
![formerror](https://cloud.githubusercontent.com/assets/896472/16081219/882840a4-3304-11e6-9d6c-570e8fa152f4.png)

The way to fix this, is to put
```js
if (error) {
  vm.hasError = true;
  Notify.handleError(error);
}
```
That should resolve the problem.